software/firmware/bootloader.py

```diff
@@ -89,7 +89,6 @@ def show_error(cls, title, message, duration=1):
         """
         oled.fill(0)
         oled.centre_text(f"--{title}--\n{message}")
-        oled.show()
```

Comment:
[optional] There's also a similar situation in the kompari script.

Reply:
Removed from bootloader & kompari.

Personally, I dislike how the `centre_text` function automatically clears the buffer at the start & then shows the buffer at the end, unlike every other oled drawing function. I didn't want to aggressively make that change here, but it might be worth a discussion at some point about making that code behave more like everything else oled-related?

Reply:
I was thinking the same thing while doing this review.

`centre_text` is amongst the oldest code in the firmware. I remember one of my first PRs, which was mostly documentation, I tried to fix the 'typo' in the spelling of center. Rory wouldn't go for it. 

Anyway, we've come a long way since then and I agree that it definitely sticks out in that API as weird.  I've put it on 'my list', but will probably open a ticket for it.

Reply:
> I tried to fix the 'typo' in the spelling of center. Rory wouldn't go for it.
I was tempted to do the same. In Canada we normally use the British spellings, but I've gotten used to using American spellings inside code just for easier API compatibility. `centre` -> `center`, `colour` -> `color`, etc... at least in anything public.

I think adding the keyword variables for the atypical behaviour is at least a good start. It exposes that the function does things differently, and it lets you be explicit about it if you need to be.